pybossa/view/projects.py

```diff
@@ -813,6 +813,156 @@ def details(short_name):
     return handle_content_type(response)
 
 
+def update_data(body, project):
+    if "project" in body:
+        key = "project"
+        data = body.get(key) or {}
+        external_config = project.info.get('ext_config') or {}
```

Comment:
seems each page has its own save option, hence else if block. if so, would it be good to have single save option that take updates from all tabs and store? Can each tab functionality be a separate function?

Reply:
Imo a save for each tab is fine, but why not reuse the already existing save endpoints? Looks like there is a lot of duplicated logic (validations), etc.

Reply:
The existing endpoints are routed to other urls instead of `~/summary` page, so I suppose I have to put duplicate code under ` @blueprint.route('/<short_name>/summary')`,  maybe one way to avoid duplicate code is to encapsulate them in a new function and call that function from each endpoints?

Reply:
Your POST request doesn't need to hit `/summary` - your frontend can make a request to  `/<short_name>/coowners` for example

Reply:
Btw most of pybossa endpoints return json when you call them with 'Content-Type: application/json', which makes it easy to reuse them to write spa-like frontends. In this case for example you could leave it to the js code to call to one of the old endpoints to fetch/save the necessary data, instead of having to rewrite backend logic to read and write data.

Reply:
ok, I agree that we can hit other endpoints to save data. but there's another problem that old endpoints use WTForm to receive and parse request data, e.g. 
` form = SearchForm(request.form)`
and now we rewrite them in Vue. So is there a good way to parse JSON data to make it a `request.form`? Or we can use if-else logic to handle two different situations.

Reply:
That should not be a problem, [this piece of code](https://github.com/bloomberg/pybossa/blob/master/pybossa/core.py#L582) is to allow you to create a WTForm instance when the content type of the request is json. See [examples in tests](https://github.com/bloomberg/pybossa/blob/master/test/test_web.py#L4000)

Reply:
discussed that we won't change the overall design for now.